capellambse/filehandler/gitlab_artifacts.py

```diff
@@ -0,0 +1,260 @@
+
+    def __resolve_project(self, project: str | int | None) -> str | int:
+        if isinstance(project, int):
+            return project
+
+        if project:
+            project = urllib.parse.quote(project, safe="")
+            info = self.__get(f"/projects/{project}")
+            return info["id"]
+
+        if project := os.getenv("CI_PROJECT_ID"):
+            return int(project)
+
+        raise TypeError("No 'project' specified, and not running in Gitlab CI")
+
+    def __resolve_job(self, job: str | int) -> int:
+        if isinstance(job, int):
+            return job
+
+        for jobinfo in self.__iterget(
+            f"/projects/{self.__project}/jobs?scope=success", max=200
+        ):
+            if (
+                jobinfo["name"] == job
+                and jobinfo["pipeline"]["ref"] == self.__branch
+                and "artifacts_file" in jobinfo
+            ):
+                return jobinfo["id"]
+
+        raise RuntimeError(f"No recent successful {job!r} job found")
+
+    def __rawget(self, url: str) -> requests.Response:
+        """Make a GET request and return the raw Response object.
+
+        Parameters
+        ----------
+        url
+            The full URL to request, including the server and protocol.
+        """
+        LOGGER.debug("GET %s", url)
+        return requests.get(
+            url, headers={"PRIVATE-TOKEN": self.__token}, timeout=30
+        )
+
+    def __get(self, path: str) -> t.Any:
+        """Make a GET request and return the decoded JSON response."""
+        response = self.__rawget(f"{self.__path}/api/v4{path}")
+        LOGGER.debug("Status: %d", response.status_code)
+        response.raise_for_status()
+        return response.json()
+
+    def __iterget(self, path: str, *, max: int = 0) -> cabc.Iterator[t.Any]:
+        """Iterate over a result set, using multiple GET requests if necessary.
+
+        This method implements the keyset-based pagination method.
+
+        Parameters
+        ----------
+        path
+            The base URL for the first request.
+        max
+            Maximum number of entries to yield. If 0, yield everything.
+        """
+        sep = ("?", "&")["?" in path]
+        next_url = (
+            f"{self.__path}/api/v4{path}{sep}pagination=keyset&per_page="
+            + str(min(100, max or 100))
+        )
+        i = 0
+        stop = max or math.inf
+
+        while True:
+            response = self.__rawget(next_url)
+            response.raise_for_status()
+            for i, object in enumerate(response.json(), start=i):
+                yield object
+                if i >= stop:
+                    return
+
+            match = RE_LINK_NEXT.fullmatch(response.headers["Link"])
+            if not match:
+                break
+            next_url = match.group(0)
+
+    def get_model_info(self) -> loader.ModelInfo:
+        return loader.ModelInfo(branch=self.__branch, url=self.__path)
```

Comment:
There is no docstring. But also, what can you write here that isn't already covered by its name.